Assets/Scripts/UserInterface/UIButton.cs

```diff
@@ -0,0 +1,70 @@
+/**
+ * Author(s): Ranjini Das
+ **/
+
+using UnityEngine;
+using UnityEngine.UI;
+using System.Collections;
+
+public class UIButton : MannBehavior
+{
+    GameObject gameObject;
+    EventController.Action action; 
+    bool isDown;
+    public delegate void performAction (EventController.Action action);  
```

Comment:
I think perform action can just be a regular function, not a delegate, comparable to CallActions() above. The Action delegate is what you're subscribing to the event